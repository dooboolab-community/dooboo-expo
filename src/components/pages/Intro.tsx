import {Button, useTheme} from 'dooboo-ui';
import {Image, View} from 'react-native';

import {IC_MASK} from '../../utils/Icons';
import IntroView from '../uis/IntroTemp';
import React from 'react';
import {RootStackNavigationProps} from '../navigations/RootStack';
import {User} from '../../types';
import {getString} from '../../../STRINGS';
import styled from '@emotion/native';
import {useAppContext} from '../../providers/AppProvider';

const Container = styled.View`
  flex: 1;
  align-self: stretch;
  overflow: scroll;
  background-color: ${({theme}) => theme.background};

  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;
`;

const ButtonWrapper = styled.View`
  position: absolute;
  flex-direction: column;
  bottom: 40px;
  width: 72%;
  align-self: center;
`;

interface Props {
  navigation: RootStackNavigationProps<'Intro'>;
}

function Intro(props: Props): React.ReactElement {
  let timer: any;

  const {setUser} = useAppContext();

  const {changeThemeType} = useTheme();
  const [isLoggingIn, setIsLoggingIn] = React.useState<boolean>(false);

  const onLogin = (): void => {
    setIsLoggingIn(true);

    timer = setTimeout(() => {
      const tempUser: User = {
        displayName: 'dooboolab',
        age: 30,
        job: 'developer',
      };

      setUser(tempUser);
      setIsLoggingIn(false);
      clearTimeout(timer);
    }, 1000);
  };

  return (
    <Container>
      <IntroView />
      <ButtonWrapper>
        <Button
          testID="btn-login"
          leftElement={
            <Image
              source={IC_MASK}
              style={{
                position: 'absolute',
                left: 12,
                width: 24,
                height: 24,
              }}
            />
          }
          loading={isLoggingIn}
          onPress={(): void => onLogin()}
          text={getString('LOGIN')}
        />
        <View style={{marginTop: 12}} />
        <Button
          testID="btn-navigate"
          onPress={(): void =>
            props.navigation.navigate('Temp', {
              param: 'GO BACK',
            })
          }
          text={getString('NAVIGATE', {name: 'Temp'})}
        />
        <View style={{marginTop: 12}} />
        <Button
          testID="btn-theme"
          onPress={(): void => changeThemeType()}
          text={getString('CHANGE_THEME')}
        />
      </ButtonWrapper>
    </Container>
  );
}

export default Intro;
