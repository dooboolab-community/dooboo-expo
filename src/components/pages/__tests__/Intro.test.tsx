import React, {ReactElement} from 'react';
import {RenderAPI, act, fireEvent, render} from '@testing-library/react-native';
import {createTestElement, createTestProps} from '../../../../test/testUtils';

import {Button} from 'dooboo-ui';
import Intro from '../Intro';
import renderer from 'react-test-renderer';

let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

describe('[Intro] screen rendering test', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Intro {...props} />);
    testingLib = render(component);
  });

  it('should render outer component and snapshot matches', () => {
    const json = renderer.create(component).toJSON();

    expect(json).toMatchSnapshot();
    expect(json).toBeTruthy();
  });

  it('should render [Dark] theme', () => {
    component = createTestElement(<Intro {...props} />, 'dark');
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it('should render [isLoading] status', () => {
    props = createTestProps({
      isLoading: true,
    });

    component = createTestElement(<Intro {...props} />, 'dark');
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it('should render [isDisabled] status', () => {
    props = createTestProps({
      isDisabled: true,
    });

    component = createTestElement(<Intro {...props} />, 'dark');
    testingLib = render(component);

    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('[Intro] Interaction', () => {
  let rendered: renderer.ReactTestRenderer;
  let root: renderer.ReactTestInstance;

  it('should simulate login when button has clicked', () => {
    testingLib = render(component);
    rendered = renderer.create(component);
    root = rendered.root;

    jest.useFakeTimers();

    const buttons = root.findAllByType(Button);

    const button = testingLib.getByTestId('btn-login');

    act(() => {
      fireEvent.press(button);
      jest.runOnlyPendingTimers();
    });

    expect(buttons[0].props.loading).toEqual(false);
  });

  it('should navigate when button has clicked', () => {
    testingLib = render(component);

    fireEvent.press(testingLib.getByTestId('btn-navigate'));

    expect(props.navigation.navigate).toHaveBeenCalledWith('Temp', {
      param: 'GO BACK',
    });
  });

  it('should change theme when button has clicked', () => {
    testingLib = render(component);

    fireEvent.press(testingLib.getByTestId('btn-theme'));
  });
});
