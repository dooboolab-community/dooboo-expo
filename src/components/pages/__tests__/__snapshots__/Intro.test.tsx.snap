// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Intro] screen rendering test should render [Dark] theme 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "black",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "lineHeight": 27,
          },
          Object {
            "marginTop": 100,
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "lineHeight": 27,
          },
          undefined,
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "lineHeight": 27,
          },
          undefined,
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "bottom": 40,
          "flexDirection": "column",
          "position": "absolute",
          "width": "72%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-login"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "white",
              "borderLeftColor": "white",
              "borderRightColor": "white",
              "borderTopColor": "white",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Image
          source={
            Object {
              "canInstrument": true,
              "createTransformer": [Function],
              "getCacheKey": [Function],
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 24,
              "left": 12,
              "position": "absolute",
              "width": 24,
            }
          }
        />
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "black",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Login
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 12,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-navigate"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "white",
              "borderLeftColor": "white",
              "borderRightColor": "white",
              "borderTopColor": "white",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "black",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Navigate to Temp
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 12,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-theme"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "white",
              "borderLeftColor": "white",
              "borderRightColor": "white",
              "borderTopColor": "white",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "black",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Change theme
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Intro] screen rendering test should render [isDisabled] status 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "black",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "lineHeight": 27,
          },
          Object {
            "marginTop": 100,
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "lineHeight": 27,
          },
          undefined,
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "lineHeight": 27,
          },
          undefined,
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "bottom": 40,
          "flexDirection": "column",
          "position": "absolute",
          "width": "72%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-login"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "white",
              "borderLeftColor": "white",
              "borderRightColor": "white",
              "borderTopColor": "white",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Image
          source={
            Object {
              "canInstrument": true,
              "createTransformer": [Function],
              "getCacheKey": [Function],
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 24,
              "left": 12,
              "position": "absolute",
              "width": 24,
            }
          }
        />
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "black",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Login
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 12,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-navigate"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "white",
              "borderLeftColor": "white",
              "borderRightColor": "white",
              "borderTopColor": "white",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "black",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Navigate to Temp
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 12,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-theme"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "white",
              "borderLeftColor": "white",
              "borderRightColor": "white",
              "borderTopColor": "white",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "black",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Change theme
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Intro] screen rendering test should render [isLoading] status 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "black",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "lineHeight": 27,
          },
          Object {
            "marginTop": 100,
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "lineHeight": 27,
          },
          undefined,
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "lineHeight": 27,
          },
          undefined,
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "bottom": 40,
          "flexDirection": "column",
          "position": "absolute",
          "width": "72%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-login"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "white",
              "borderLeftColor": "white",
              "borderRightColor": "white",
              "borderTopColor": "white",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Image
          source={
            Object {
              "canInstrument": true,
              "createTransformer": [Function],
              "getCacheKey": [Function],
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 24,
              "left": 12,
              "position": "absolute",
              "width": 24,
            }
          }
        />
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "black",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Login
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 12,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-navigate"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "white",
              "borderLeftColor": "white",
              "borderRightColor": "white",
              "borderTopColor": "white",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "black",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Navigate to Temp
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 12,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-theme"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderBottomColor": "white",
              "borderLeftColor": "white",
              "borderRightColor": "white",
              "borderTopColor": "white",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "black",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Change theme
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`[Intro] screen rendering test should render outer component and snapshot matches 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 18,
            "lineHeight": 27,
          },
          Object {
            "marginTop": 100,
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 18,
            "lineHeight": 27,
          },
          undefined,
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 18,
            "lineHeight": 27,
          },
          undefined,
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "bottom": 40,
          "flexDirection": "column",
          "position": "absolute",
          "width": "72%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-login"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "borderBottomColor": "black",
              "borderLeftColor": "black",
              "borderRightColor": "black",
              "borderTopColor": "black",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Image
          source={
            Object {
              "canInstrument": true,
              "createTransformer": [Function],
              "getCacheKey": [Function],
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 24,
              "left": 12,
              "position": "absolute",
              "width": 24,
            }
          }
        />
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "white",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Login
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 12,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-navigate"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "borderBottomColor": "black",
              "borderLeftColor": "black",
              "borderRightColor": "black",
              "borderTopColor": "black",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "white",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Navigate to Temp
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 12,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="btn-theme"
    >
      <View
        onLayout={[Function]}
        outlined={false}
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "borderBottomColor": "black",
              "borderLeftColor": "black",
              "borderRightColor": "black",
              "borderTopColor": "black",
            },
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 10,
              },
              Array [
                undefined,
                false,
                undefined,
              ],
            ],
          ]
        }
        testID="button-container"
        type="primary"
      >
        <Text
          outlined={false}
          style={
            Array [
              Object {
                "color": "white",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
          type="primary"
        >
          Change theme
        </Text>
      </View>
    </View>
  </View>
</View>
`;
